src/ripple/app/tx/impl/SusPay.cpp

```diff
@@ -264,45 +334,37 @@ SusPayFinish::preflight (PreflightContext const& ctx)
     if (!isTesSuccess (ret))
         return ret;
 
-    if (ctx.tx[~sfMethod])
+    auto const cb = ctx.tx[~sfCondition];
+    auto const fb = ctx.tx[~sfFulfillment];
+
+    // If you specify a condition, then you must also specify
+    // a fulfillment.
+    if (static_cast<bool>(cb) != static_cast<bool>(fb))
+        return temMALFORMED;
+
+    if (cb && fb)
     {
-        // Condition
-        switch(ctx.tx[sfMethod])
-        {
-        case 1:
+        auto& router = ctx.app.getHashRouter();
+
+        auto const id = ctx.tx.getTransactionID();
+        auto const flags = router.getFlags (id);
+
+        // If we already checked the fulfillment and found
+        // it to be bad, don't bother checking again:
+        if (flags & SF_CONDBAD)
```

Comment:
`SF_CONDBAD` and `SF_CONDGOOD` are defined in `apply.cpp`. We don't have access to the definitions here. Move the defs to a header file?